### PR TITLE
Remove ``l_abs_step_tol"

### LINKED ISSUE #11970: l_abs_step_tol is completely broken
## Rationale
If the user supplies any positive number for `l_abs_step_tol` in `Executioner`, linear solves always converge in 1 iteration, even if the linear solve is nowhere even close to convergence.

Furthermore, the check in `FEProblemBase::checkLinearConvergence` is completely wrong anyway; it does
```
| ||r^{old}|| - ||r^{new}|| | < l_abs_step_tol
```
This is just wrong. For instance, the two residual vectors could be very different, yet have the same norms, so step convergence would be detected falsely.  The correct implementation is
```
||r^{old} - r^{new}|| < l_abs_step_tol
```

## Description
In `FEProblemBase::checkLinearConvergence`, print out `rnorm` and `system._last_rnorm`. You'll see that on the second call to this function in each linear solve, `rnorm` and `system._last_rnorm` are exactly equal, which seems to suggest that `system._last_rnorm` is updated with the new `rnorm` *before* calling this function.

There appears to be only 1 input file in MOOSE where `l_abs_step_tol` is even used: `modules/combined/examples/mortar/mortar_gradient.i`. You'll notice that those linear solves converge in 1 iteration.

## Impact
Fix broken linear convergence monitoring.

### PR DESCRIPTION
**Pull request by fdkong**
Remove `l_abs_step_tol` that is confusing. Instead, we should add linear absolute tolerance 
that is consistent with PETSc.

Closes #11970

### CODE REVIEW

**Comment by fdkong** [comment]
Curious  about how many tests will fail

**Comment by fdkong** [comment]
Oh my god, RattleSnake copies the whole moose so that we can not delete it right away. Let us deprecate it first, and then fix RS and remove it.

**Review by permcody — APPROVED** [review]

**Comment by YaqiWang** [comment]
Just noticed a new parameter is added `l_abs_tol` here. Should we check this value should be always smaller than `nl_abs_tol`? It could be a little dangerous.

**Comment by fdkong** [comment]
I do not understand why we need to have `l_abs_tol` smaller than `nl_abs_tol`. They could be totally unrelated. 

Yes I will add a test.

**Comment by YaqiWang** [comment]
Typically linear residual norm of last linear iteration is higher than the nonlinear residual norm right after it. So if we set linear residual norm with `l_abs_tol`, we could quit linear solving too early? Just feel strange when users accidentally set `l_abs_tol` larger than `nl_abs_tol`. I think we can just remove `l_abs_tol` and keep it as an advanced PETSc option.

**Comment by fdkong** [comment]
No. They are unrelated unless the problem is very close to linear or the problem is just linear. Another thing, if the left preconditioner is used, then the linear residual is totally unrelated to the nonlinear residual. 

I would like to keep it as is.

**Comment by YaqiWang** [comment]
Looks like you did not get what I was saying. Yes, linear residual and nonlinear residual are two different thing. If both l_abs_tol and nl_abs_tol are set, our PJFNK solver's *goal* is to satisfy `nl_abs_tol`. `l_abs_tol` is for controlling linear solves in the Newton iteration. The problem is that large `l_abs_tol` could badly affecting our goal of achieving `nl_abs_tol`.

File: framework/src/utils/PetscSupport.C [hunk 1]
```diff
@@ -581,7 +581,7 @@ petscSetKSPDefaults(FEProblemBase & problem, KSP ksp)
   auto & es = problem.es();
 
   PetscReal rtol = es.parameters.get<Real>("linear solver tolerance");
-  PetscReal atol = es.parameters.get<Real>("linear solver absolute step tolerance");
+  PetscReal atol = es.parameters.get<Real>("linear solver absolute tolerance");
```
Comment (YaqiWang) [inline framework/src/utils/PetscSupport.C]: This bug was introduced by @friedmud in the actual Euler time stepper. Before that time, I think `l_abs_step_tol` used to work. It is unfortunate we have to remove this parameter.

**Pull request merged**